metaflow/plugins/argo/argo_workflows.py

```diff
@@ -819,6 +819,20 @@ def _container_templates(self):
             # liked to inline this ContainerTemplate and avoid scanning the workflow
             # twice, but due to issues with variable substitution, we will have to
             # live with this routine.
+            resource_requests={
+                "cpu": str(resources["cpu"]),
+                "memory": "%sM" % str(resources["memory"]),
+                "ephemeral-storage": "%sM" % str(resources["disk"]),
+            }
+
+            resource_limits={
+                "%s.com/gpu".lower()
+                % resources["gpu_vendor"]: str(resources["gpu"])
+                for k in [0]
+                if resources["gpu"] is not None
+            }
+            resource_limits.update(resource_requests)
```

Comment:
We do this because we want Guaranteed QoS?

Reply:
Yes, but also default requests from argo plug-in is higher than default limits in our cluster, and argo plugin doesn't set limits without this change resulting in a request > limit error on K8s.

Reply:
Sounds good, this seems like something we can definitely contribute back upstream under a "Guaranteed QoS" feature potentially right?

Reply:
I guess we can make it a case that non guaranteed pods are hard to work with for batch workflow, and propose the same change on netflix/metaflow. Alternatively this can be a QoS config from zg plug-in.

Reply:
Yeah makes sense, I think it's certainly true for memory given current OS-level limitations but for CPU I think it's less clear cut but I do like the idea of proposing this change upstream as the default though and see what they say :) 

Obviously this is a minor change but I guess the philosophy I'm hopeful we can adopt is to move a little slower this time with less forking. May not be possible though unfortunately.

Reply:
> Alternatively this can be a QoS config from zg plug-in.

Can you please add a comment to link to the JIRA epic/task to create a Metaflow Platform Plugin feature and zg plug-in?